File: fly-promo-publisher-flyWU/src/main/java/io/github/riojano0/flypromoflyWU/controller/FlyPromoController.java
package io.github.riojano0.flypromoflyWU.controller;

import io.github.riojano0.flypromoflyWU.model.FlyWuFlyPromo;
import io.github.riojano0.flypromoflyWU.service.FlyPromoService;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RestController;

import lombok.extern.slf4j.Slf4j;

@Slf4j
@RestController
public class FlyPromoController {

   private final FlyPromoService flyPromoService;

   @Autowired
   public FlyPromoController(FlyPromoService flyPromoService) {
      this.flyPromoService = flyPromoService;
   }

   @GetMapping
   public FlyWuFlyPromo publishPromoDummy() {
      return flyPromoService.generateFlyPromo();
   }

}

File: fly-promo-publisher-fly2EU/src/main/kotlin/io/github/riojano0/flypromofly2EU/controller/FlyPromoController.kt
package io.github.riojano0.flypromofly2EU.controller

import io.github.riojano0.flypromokafka.model.FlyPromo
import io.github.riojano0.flypromofly2EU.service.FlyPromoService
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RestController


@RestController
class FlyPromoController @Autowired constructor(val flyPromoService: FlyPromoService) {

    @GetMapping
    fun publishPromoDummy(): FlyPromo {
        return flyPromoService.generateFlyPromo()
    }

}